frontend/src/pages/projects/screens/detail/WorkspacesList.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { Button, Flex, FlexItem, Stack, StackItem, Title } from '@patternfly/react-core';
+import EmptyDetailsList from './EmptyDetailsList';
+
+const WorkspacesList: React.FC = () => {
+  return (
+    <Stack hasGutter>
+      <StackItem>
+        <Flex>
+          <FlexItem>
+            <Title id="data-science-workspaces" headingLevel="h4" size="xl">
+              Data science workspaces
+            </Title>
+          </FlexItem>
+          <FlexItem>
+            <Button variant="secondary">Create data science workspace</Button>
+          </FlexItem>
+        </Flex>
+      </StackItem>
+      <StackItem>
+        <EmptyDetailsList
+          title="No data science workspaces"
+          description="To get started, create a data science workspace."
+        />
+      </StackItem>
+    </Stack>
```

Comment:
No way to DRY this? You have a lot of repeat boilerplate code.

What about something like:
```
<DetailsSection
  id={...the enum you could create...}
  title="..."
  actions={[
    <Button key="create" variant="secondary">...</Button>,
  ]}
>
  <EmptyDetailsList title="..." description="..." />
</DetailsSection>
```

When we get into the data states, we can just do a ternary around the `EmptyDetailsList` component and render the data (or maybe we have a wrapper there too, and this is just a prop to THAT component). Anyway we cut this, we should look to be more detailed with our intentions.